tests/mocks/mock_tool/tool.py

```diff
@@ -52,6 +53,15 @@ def test_str_output(self, value: dict) -> str:
     def test_no_schema(self, value: dict) -> str:
         return "no schema"
 
+    @activity(
+        config={
+            "description": "test description",
+            "schema": lambda _self: Schema(_self.custom_schema, description="Test input"),
+        }
+    )
+    def test_callable_schema(self) -> TextArtifact:
+        return TextArtifact("ack")
+
```

Comment:
Could you add one other test where the callable passed is actually another method on the tool?

Reply:
Added/updated tests to show this.